Assets/Scripts/DataTypes/Vertices/EndPoint.cs

```diff
@@ -10,31 +11,46 @@ public class EndPoint : Vertex<EndPoint, EndPointBehaviour>
         private GameObject _carPrefab { get; }
         // ticks before a car spawns on a lane (index)
         private Frequencies _frequencies { get; }
-        public Dictionary<IVertex, List<Edge>> routingTable { get; } = new Dictionary<IVertex, List<Edge>>();
+        // weighed choosing of vertices to route to
+        private RouteProbabilities _routeProbabilities;
+        private int[] _weights { get; }
+        public Dictionary<IVertex, List<RouteSegment>> routingTable { get; } = new Dictionary<IVertex, List<RouteSegment>>();
+        private static readonly HashSet<LaneType> _ONLY_THROUGH = new HashSet<LaneType> {LaneType.Through};
 
-        public EndPoint(Edge edge, GameObject carPrefab, Frequencies frequencies) : base(edge)
+        public EndPoint(Edge edge, GameObject carPrefab, Frequencies frequencies, int[] weights) : base(edge)
         {
             _edge = edge;
             _carPrefab = carPrefab;
             _frequencies = frequencies;
+            _weights = weights;
+
+            if (edge.incomingLanes.Any(lane => lane.types.Equals(_ONLY_THROUGH)))
+                throw new Exception("All lanes going into an EndPoint have to be of type Through");
+        }
+
+        public void SetWeights()
+        {
+            _routeProbabilities = new RouteProbabilities(_weights, routingTable.Where(kvp => kvp.Value != null).Select(kvp => kvp.Key).ToList());
         }
```

Comment:
gefällt mir irgendwie nicht ganz, dass diese Methode in EndPoint drin ist und EndPoint nur funktioniert, wenn es in Pathfinding nicht vergessen wird sie aufzurufen. Gib bestimmt ne bessere Methode.
Außerdem: #84, aber das ist denke ich klar

Reply:
Wir ja nur einmal aufgerufen und dabei werden auch alle aufgerufen. Von daher kein Problem